gnocchi-core/src/main/scala/org/bdgenomics/gnocchi/algorithms/siteregression/LinearSiteRegression.scala

```
@@ -38,81 +39,70 @@ trait LinearSiteRegression[VM <: LinearVariantModel[VM]] extends SiteRegression[
+      val genos = matX(::, 0)
+      val deviations = genos - mean(genos)
+      val ssDeviations = deviations.t * deviations
 
       // compute the regression parameters standard errors
-      val standardErrors = ols.estimateRegressionParametersStandardErrors()
+      val betaVariance = diag(inv(matX.t * matX))
+      val sigma = residuals.t * residuals / (matX.rows - matX.cols)
+      val standardErrors = sqrt(sigma * betaVariance)
 
       // get standard error for genotype parameter (for p value calculation)
       val genoSE = standardErrors(1)
 
       // test statistic t for jth parameter is equal to bj/SEbj, the parameter estimate divided by its standard error
-      val t = beta(1) / genoSE
+      val t = result(1) / genoSE
 
       /* calculate p-value and report:
         Under null hypothesis (i.e. the j'th element of weight vector is 0) the relevant distribution is
         a t-distribution with N-p-1 degrees of freedom. (N = number of samples, p = number of regressors i.e. genotype+covariates+intercept)
         https://en.wikipedia.org/wiki/T-statistic
       */
-      val residualDegreesOfFreedom = genotypes.numValidSamples - phenotypesLength - 1
-      val tDist = new TDistribution(residualDegreesOfFreedom)
-      val pvalue = 2 * tDist.cumulativeProbability(-math.abs(t))
-      val logPValue = log10(pvalue)
+      val residualDegreesOfFreedom = matX.rows - matX.cols - 1
+      val tDist = StudentsT(residualDegreesOfFreedom)
+      val pValue = 2 * tDist.cdf(-math.abs(t))
+      val logPValue = log10(pValue)
 
       LinearAssociation(
         ssDeviations,
         ssResiduals,
         genoSE,
         t,
         residualDegreesOfFreedom,
-        pvalue,
-        beta.toList,
+        pValue,
+        result.data.toList,
         genotypes.numValidSamples)
     } catch {
-      case _: breeze.linalg.MatrixSingularException => {
-        throw new SingularMatrixException()
-      }
+      case _: MatrixSingularException =>
```

In the `LogisticSiteRegression` rewrite, I decided it makes more sense to have the `applyToSite` methods throw the singular matrix exceptions and have the higher level apply methods do the error handling. We can remove the try/catch block here and just let the method throw any errors, then deal with them in the apply method for consistency with `LogisticSiteRegression`